db/backup/recovery_test.py

```diff
@@ -0,0 +1,246 @@
+""" A script to test that tests the validity of a database backup
+in the event of recovery.
+NOTE: DO NOT RUN THIS SCRIPT ON A PRODUCTION SERVER.
+It will drop the local mysql database after each run.
+
+"""
+
+import os
+import unittest
+import subprocess
+from collections import namedtuple
+from google.cloud import storage
+import mysql.connector
+
+
+BACKUP_BUCKET = 'cpg-sm-backups'
+LOCAL_BACKUP_FOLDER = 'latest_backup'
+DATABASE = 'sm_production'
+
+ConnectionDetails = namedtuple('ConnectionDetails', 'address user')
+
+
+class TestDatabaseBackup(unittest.TestCase):
+    """ Testing validity of DB backup"""
+
+    @classmethod
+    def setUpClass(cls):
+        """ Pull the backup file, and restore the database."""
+        backup_folder = pull_latest_backup()
+        get_timestamp(backup_folder)
+        # Pull the latest backup
+        subprocess.run(['mkdir', LOCAL_BACKUP_FOLDER], check=True)
+        subprocess.run(
+            [
+                'gsutil',
+                '-m',
+                'cp',
+                '-r',
+                f'gs://cpg-sm-backups/{backup_folder}',
+                LOCAL_BACKUP_FOLDER,
+            ],
+            check=True,
+        )
+        # Prepare the backup for restoration
+        subprocess.run(
+            [
+                'mariabackup',
+                '--prepare',
+                f'--target-dir={LOCAL_BACKUP_FOLDER}/{backup_folder}',
+            ],
+            check=True,
+        )
+        # Stop the MariaDB server
+        subprocess.run(['sudo', 'systemctl', 'stop', 'mariadb'], check=True)
+
+        if os.path.isdir('/var/lib/mysql/'):
+            print('Backup cannot be performed unless /var/lib/mysql is empty.')
```

Comment:
Should `isdir` always return true, or are you checking if the folder doesn't exist?

Reply:
Checking that the folder doesn't exist. In hindsight, we should catch and exit here. I'll amend :)